src/Forms/Form.php

```diff
@@ -125,6 +137,14 @@ protected function processComponentDefinition(array $componentJson, $path = ''):
                 $component->setStorageDriver(resolve($storageDriver));
             }
 
+//            Because component is instantiated above, its constructor can't call methods that use resourceRegistry
+            if (is_subclass_of($component, ResourceValues::class)) {
+                $component->setResourceRegistry($this->resourceRegistry);
+                if ($component->type() == 'select' && $component->dataSource() == Select::DATA_SRC_RESOURCE) {
```

Comment:
Would it make more sense to move this into the `setResourceRegistry` method? If it's a component that uses resources, it'll sorts itself without needing to make an extra call.